### PR TITLE
Follow-up cleanup of GitHub Actions migration

### PR DESCRIPTION
**Pull request by rnro**
Follow-up cleanup of GitHub Actions migration.

### Motivation:

Make use of more common GitHub Actions migration workflows and repository style changes.

### Modifications:

* Unit tests workflows run redis in a separate services container to speed up unit tests
* Introduce `main.yml` which runs workflows on each commit to main and periodically to catch any regressions in merges or from upstream.
* Remove the docker files which are no longer used and contained outdated pipelines.
* Add Cxx interoperability checks

### Result:

More in common with other GitHub Actions adoptions.

### CODE REVIEW

**Review by FranzBusch — APPROVED** [review]

**Review by FranzBusch — APPROVED** [review]

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        description: "Boolean to enable the Linux nightly main Swift version matrix job. Defaults to true."
+        default: true
+      linux_nightly_main_arguments_override:
+        type: string
+        description: "The arguments passed to swift test in the Linux nightly main Swift version matrix job."
+        default: ""
+
+jobs:
+  unit-tests:
+    name: Unit tests
```
Comment (FranzBusch) [inline .github/workflows/unit_tests.yml:49]:
```suggestion
    name: inux (${{ matrix.swift.swift_version }})
```

File: scripts/generate_contributors_list.sh [hunk 2]
```diff
@@ -1,53 +0,0 @@
-#!/bin/bash
```
Comment (fabianfett) [inline scripts/generate_contributors_list.sh:1]: Why was this file removed?
